weblate/api/views.py

```diff
@@ -982,6 +984,24 @@ def download_archive(self, request, **kwargs):
         )
 
 
+class MemoryViewSet(viewsets.ModelViewSet, DestroyModelMixin):
+    """Memory API."""
+
+    queryset = Memory.objects.none()
+    serializer_class = MemorySerializer
+
+    def get_queryset(self):
+        return Memory.objects.order_by("id")
```

Comment:
Either all this view needs to be guarded by permission, or the list filtered to list only allowed entries.

As the filtering will be quite complex, I'd stick with allowing listing to superusers only.